packages/next-app/src/app/metadata/[poolId]/page.tsx

```diff
@@ -1,35 +1,50 @@
 "use client";
 
-import { useRef, useState } from "react";
+import { useEffect, useRef, useState } from "react";
+import { useContractWrite, usePrepareContractWrite } from "wagmi";
 
 import { Button } from "../../../components";
+import useDebounce from "../../../hooks/useDebounce";
 import metadataGql from "../../../lib/poolMetadataGql";
+import { poolMetadataRegistryABI } from "../../../wagmi/generated";
 
 export default function Page({ params }: { params: { poolId: string } }) {
   const textField = useRef<HTMLTextAreaElement>(null);
 
-  const [metadata, setMetadata] = useState(`[
-    {
-      "name": "Pool Address",
-      "type": "address",
-      "desc": "The address of the smart contract that implements the exchange pool",
-      "value": "0x7a250d5630B4cF539739dF2C5dAcb4c659F2488D"
-    },
-    {
-      "name": "Pool link",
-      "type": "URL",
-      "desc": "The address of the smart contract that implements the exchange pool",
-      "value": "https://github.com/"
-    }
-  ]`);
+  const [metadata, setMetadata] = useState();
 
   const [pinned, setPinned] = useState("");
+  const [metadataCID, setMetadataCID] = useState("");
+  const debouncedCID = useDebounce(metadataCID, 500);
 
   const { data: pools } = metadataGql.useMetadataPool({
     poolId: params.poolId,
   });
   const metadataPool = pools?.pools[0];
 
+  useEffect(() => {
+    if (metadataPool?.metadataCID) {
+      fetch(`${process.env.NEXT_PUBLIC_PINATA_GW}/${metadataPool?.metadataCID}`)
+        .then((res) => {
+          return res.json();
+        })
+        .then((data) => {
+          setMetadata(JSON.stringify(data, null, 2));
+        });
+    }
+  }, [metadataPool?.metadataCID]);
+
+  const { config } = usePrepareContractWrite({
+    address: "0xebfadf723e077c80f6058dc9c9202bb613de07cf",
+    abi: poolMetadataRegistryABI,
+    functionName: "setPoolMetadata",
+    args: [`0x${params.poolId.slice(2)}`, debouncedCID],
+    enabled: Boolean(debouncedCID),
+  });
```

Comment:
wouldn't it be easier to use the generated `usePreparePoolMetadataRegistrySetPoolMetadata` and `usePoolMetadataRegistrySetPoolMetadata`?

`#must` use `usePreparePoolMetadataRegistrySetPoolMetadata` since it'll

Reply:
Nice one! I didn't know that ✌🏻

Reply:
it's generated by `wagmi` from our project 😉